### ISSUE TITLE
[REVIEW]: EoN (Epidemics on Networks), software for simulation, analytic approximation, and analysis of epidemics on networks

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#1731]
**Submitting author:** @joelmiller (<a href="http://orcid.org/0000-0003-4426-0405">Joel Miller</a>)
**Repository:** <a href="https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks" target ="_blank">https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks</a>
**Version:** v1.1
**Editor:** @lpantano
**Reviewers:** @acolum, @pholme, @hagberg
**Archive:** <a href="https://doi.org/10.5281/zenodo.3572756" target="_blank">10.5281/zenodo.3572756</a>

## Status

[![status](https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5/status.svg)](https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5)

Status badge code:

```
HTML: <a href="https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5"><img src="https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5/status.svg"></a>
Markdown: [![status](https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5/status.svg)](https://joss.theoj.org/papers/d00ea63dc7a7c56cf73852229031b8f5)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) by leaving comments in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

## Reviewer instructions & questions

@acolum & @pholme, please carry out your review in this issue by updating the checklist below. If you cannot edit the checklist please:
1. Make sure you're logged in to your GitHub account
2. Be sure to accept the invite at this URL: https://github.com/openjournals/joss-reviews/invitations

The reviewer guidelines are available here: https://joss.readthedocs.io/en/latest/reviewer_guidelines.html. Any questions/concerns please let @lpantano know.

✨ **Please try and complete your review in the next two weeks** ✨ 

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@joelmiller) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [ ] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [ ] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [ ] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks">repository url</a>?
- [ ] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [ ] **Contribution and authorship:** Has the submitting author (@joelmiller) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [ ] **Installation:** Does installation proceed as outlined in the documentation?
- [ ] **Functionality:** Have the functional claims of the software been confirmed?
- [ ] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [ ] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [ ] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [ ] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [ ] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [ ] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [ ] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [ ] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [ ] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [ ] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

### Conflict of interest

- [x] As the reviewer I confirm that I have read the [JOSS conflict of interest policy](https://github.com/openjournals/joss/blob/master/COI.md) and that there are no conflicts of interest for me to review this work.

### Code of Conduct

- [x] I confirm that I read and will adhere to the [JOSS code of conduct](https://joss.theoj.org/about#code_of_conduct).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Contribution and authorship:** Has the submitting author (@joelmiller) made major contributions to the software? Does the full list of paper authors seem appropriate and complete?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** If there are any performance claims of the software, have they been confirmed? (If there are no claims, please check off this item.)

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g., API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the functionality of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Summary:** Has a clear description of the high-level functionality and purpose of the software for a diverse, non-specialist audience been provided?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **State of the field:** Do the authors describe how this software compares to other commonly-used packages?
- [x] **Quality of writing:** Is the paper well written (i.e., it does not require editing for structure, language, or writing quality)?
- [x] **References:** Is the list of references complete, and is everything cited appropriately that should be cited (e.g., papers, datasets, software)? Do references in the text use the proper [citation syntax]( https://rmarkdown.rstudio.com/authoring_bibliographies_and_citations.html#citation_syntax)?

**Comment by whedon** [comment]
Hello human, I'm @whedon, a robot that can help you with some common editorial tasks. @acolum, @pholme it looks like you're currently assigned to review this paper :tada:.

**:star: Important :star:**

If you haven't already, you should seriously consider unsubscribing from GitHub notifications for this (https://github.com/openjournals/joss-reviews) repository. As a reviewer, you're probably currently watching this repository which means for GitHub's default behaviour you will receive notifications (emails) for all reviews 😿

To fix this do the following two things:

1. Set yourself as 'Not watching' https://github.com/openjournals/joss-reviews:

![watching](https://cloud.githubusercontent.com/assets/4483/20250593/64d7ce48-a9de-11e6-9225-d3dfb3e48e68.png)

2. You may also like to change your default settings for this watching repositories in your GitHub profile here: https://github.com/settings/notifications

![notifications](https://cloud.githubusercontent.com/assets/4483/20250604/79eb3144-a9de-11e6-9e51-0fe940966689.png)


For a list of things I can do to help you, just type:

```
@whedon commands
```

For example, to regenerate the paper pdf after making changes in the paper's md or bib files, type:

```
@whedon generate pdf
```

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by acolum** [comment]
Just finished my review - everything looks great!

**Comment by lpantano** [comment]
Thank you @acolum! 👋 @pholme let me know if you need something else to complete the review.

**Comment by lpantano** [comment]
👋 @pholme, are you still able to review this? Let me know.

**Comment by lpantano** [comment]
@whedon add @hagberg as reviewer

**Comment by whedon** [comment]
OK, @hagberg is now a reviewer

**Comment by lpantano** [comment]
@hagberg how is the review going?

**Comment by lpantano** [comment]
Hi @hagberg, did you end up opening the issues to ask the authors to change issues you saw during the review? If yes, can you link them here? Thanks!

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
PDF failed to compile for issue #1731 with the following error: 

 Error reading bibliography ./paper.bib (line 39, column 1):
unexpected "y"
expecting space, ",", white space or "}"
Error running filter pandoc-citeproc:
Filter returned error status 1
Looks like we failed to compile the PDF

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
Hi @lpantano - Do you know if there is any way to embed a video into the pdf file?  One of the final things I want to show is a tool to build an animation, but at present I can't find any way to embed it.

**Comment by joelmiller** [comment]
@hagberg 

I've added the functional testing (created by @tinghf )  into the repository under EoN/tests

I've also added a discussion of related packages into the paper.  

Let me know of anything else you'd like improved.

**Comment by hagberg** [comment]
Looks good.  Issues are closed.

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by lpantano** [comment]
> Hi @lpantano - Do you know if there is any way to embed a video into the pdf file? One of the final things I want to show is a tool to build an animation, but at present I can't find any way to embed it.

I will need to ask about this, not sure this is possible.

**Comment by danielskatz** [comment]
I’m fairly sure the answer is no

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@lpantano @danielskatz 

Okay - I've removed the animation from the paper.md file.  Should be good to go now.

**Comment by lpantano** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.6084/m9.figshare.1164194 is OK

MISSING DOIs

- https://doi.org/10.1016/j.cpc.2017.06.007 may be missing for title: Optimized Gillespie algorithms for the simulation of Markovian epidemic processes on large and heterogeneous networks
- https://doi.org/10.1090/s0002-9947-1945-0013857-4 may be missing for title: Markoff chains–denumerable case
- https://doi.org/10.18637/jss.v084.i08 may be missing for title: EpiModel: An R Package for Mathematical  Modeling of Infectious Disease over Networks
- https://doi.org/10.1093/comnet/cnv021 may be missing for title: Complex contagions and hybrid phase transitions
- https://doi.org/10.1101/297267 may be missing for title: FAVITES: simultaneous simulation of transmission networks, phylogenetic trees and sequences
- https://doi.org/10.1186/s13362-019-0058-7 may be missing for title: A monotonic relationship between the variability of the infectious period and final size in pairwise epidemic modelling
- https://doi.org/10.1515/9781400841356.497 may be missing for title: A simple model of global cascades on random networks
- https://doi.org/10.1086/jar.33.4.3629752 may be missing for title: An information flow model for conflict and fission in small groups

INVALID DOIs

- None
```

**Comment by lpantano** [comment]
@joelmiller, it seems there are missing DOIs, can you double check this? Thanks!

**Comment by joelmiller** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.physa.2006.06.018 is OK
- 10.1016/j.cpc.2017.06.007 is OK
- 10.1090/S0002-9947-1945-0013857-4 is OK
- 10.18637/jss.v084.i08 is OK
- 10.1007/978-3-319-50806-1 is OK
- 10.1021/j100540a008 is OK
- 10.25080/issn.2575-9752 is OK
- 10.1093/comnet/cnv021 is OK
- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.1093/bioinformatics/bty921 is OK
- 10.6084/m9.figshare.1164194 is OK
- 10.1103/PhysRevLett.103.058701 is OK
- 10.1186/s13362-019-0058-7 is OK
- 10.1073/pnas.082090499 is OK
- 10.1086/jar.33.4.3629752 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.physa.2006.06.018 is OK
- 10.1016/j.cpc.2017.06.007 is OK
- 10.1103/PhysRevE.96.022301 is OK
- 10.1090/S0002-9947-1945-0013857-4 is OK
- 10.18637/jss.v084.i08 is OK
- 10.1021/j100540a008 is OK
- 10.1103/PhysRevE.93.042316 is OK
- 10.25080/issn.2575-9752 is OK
- 10.1007/978-3-319-50806-1 is OK
- 10.1063/1.5010002 is OK
- 10.1103/PhysRevE.87.060801 is OK
- 10.1093/comnet/cnv021 is OK
- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.1093/bioinformatics/bty921 is OK
- 10.6084/m9.figshare.1164194 is OK
- 10.1103/PhysRevLett.103.058701 is OK
- 10.1186/s13362-019-0058-7 is OK
- 10.1073/pnas.082090499 is OK
- 10.1086/jar.33.4.3629752 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by joelmiller** [comment]
@lpantano - I think this is good to go now.  I've updated the references and gone through the paper a couple times.

**Comment by lpantano** [comment]
thank you! Last step is to create a Zenodo archive: The title needs to be the same than in the paper, together with the authors. When you are ready you can paste here the Zenodo link. Thanks!

**Comment by joelmiller** [comment]
That's actually a bit problematic.  The software was developed initially alongside a textbook I was writing, so it sits within https://github.com/springer-math.  I've sent a request to give Zenodo the necessary access.  But the editor who set up that site has left, and I'm not even sure that anyone will get the message.

Thoughts?  The obvious idea would be to create a new repository.  But I prefer to move things around as little as possible.

**Comment by joelmiller** [comment]
@lpantano 

That's actually a bit problematic. The software was developed initially alongside a textbook I was writing, so it sits within https://github.com/springer-math (which I don't own). I've sent a request to give Zenodo the necessary access. But the editor who set up that site has left the company, and I'm not even sure that anyone will get the message.  It looks like it hasn't been used since he left (over a year ago).

Thoughts? The obvious idea would be to create a new repository. But I prefer to move things around as little as possible.

**Comment by danielskatz** [comment]
You can simply tar the repo and manually deposit it in Zenodo (or figshare, or any other archival repository that will accept it and provide a DOI)

This is what the GitHub - Zenodo link automates, but you don't have to use the automatic mechanism

**Comment by joelmiller** [comment]
@danielskatz @lpantano 

2 things;

- I've updated the version to 1.1 (from 1.0.8)
- I've reserved the DOI 10.5281/zenodo.3572756.  I'd like to know the final paper reference before I finalize the submission to Zenodo.  Will that be feasible?  Otherwise I'll just go ahead and submit it.

**Comment by danielskatz** [comment]
It will be https://doi.org/10.21105/joss.01731

**Comment by lpantano** [comment]
@joelmiller, let me know when this is ready and I can update here the information.

**Comment by joelmiller** [comment]
@lpantano Here you go: https://zenodo.org/record/3572756

**Comment by joelmiller** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List Whedon's capabilities
@whedon commands

# List of editor GitHub usernames
@whedon list editors

# List of reviewers together with programming language preferences and domain expertise
@whedon list reviewers

EDITORIAL TASKS

# Compile the paper
@whedon generate pdf

# Compile the paper from alternative branch
@whedon generate pdf from branch custom-branch-name

# Ask Whedon to check the references for missing DOIs
@whedon check references

```

**Comment by lpantano** [comment]
@joelmiller, can you match the title from the zenodo to the title in the paper and the authors names exactly as they appear in the paper? I hope that is not supper difficult to do. Thanks!

**Comment by joelmiller** [comment]
@lpantano, I've made those changes.  The only remaining inconsistency is that after updating changes I decided to rename this as v1.1.  I don't see the version appearing anywhere in the proof so it's probably not an issue, but I do see that in the metadata in the first post on this thread it gives the version number.  So if it matters we should update, but if not, I think we're done.

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by lpantano** [comment]
@whedon set 10.5281/zenodo.3572756 as archive

**Comment by whedon** [comment]
OK. <a href="https://doi.org/10.5281/zenodo.3572756" target="_blank">10.5281/zenodo.3572756</a> is the archive.

**Comment by lpantano** [comment]
@whedon set v1.1 as version

**Comment by whedon** [comment]
OK. v1.1 is the version.

**Comment by lpantano** [comment]
@openjournals/joss-eics, I think this is good to get an eye from you.

**Comment by arfon** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.physa.2006.06.018 is OK
- 10.1016/j.cpc.2017.06.007 is OK
- 10.1103/PhysRevE.96.022301 is OK
- 10.1090/S0002-9947-1945-0013857-4 is OK
- 10.18637/jss.v084.i08 is OK
- 10.1021/j100540a008 is OK
- 10.1103/PhysRevE.93.042316 is OK
- 10.25080/issn.2575-9752 is OK
- 10.1007/978-3-319-50806-1 is OK
- 10.1063/1.5010002 is OK
- 10.1103/PhysRevE.87.060801 is OK
- 10.1093/comnet/cnv021 is OK
- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.1093/bioinformatics/bty921 is OK
- 10.6084/m9.figshare.1164194 is OK
- 10.1103/PhysRevLett.103.058701 is OK
- 10.1186/s13362-019-0058-7 is OK
- 10.1073/pnas.082090499 is OK
- 10.1086/jar.33.4.3629752 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1186

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1186, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by arfon** [comment]
@lpantano @joelmiller - this paper is _very_ long for JOSS (24 pages). From a quick review, there look to been a substantial amount of documentation (e.g. code examples) that would be more appropriate in supporting documentation.

What do you think about moving this content into the docs?

**Comment by labarba** [comment]
In fact, the [JOSS documentation](https://joss.readthedocs.io/en/latest/submitting.html) stipulates that papers should be _short_: "The paper should be between 250-1000 words." And also,

> a “full length” paper is not permitted, and software documentation such as API (Application Programming Interface) functionality should not be in the paper and instead should be outlined in the software documentation.

TL;DR: JOSS does not publish long-form papers.

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@whedon generate pdf

**Comment by whedon** [comment]
```
Attempting PDF compilation. Reticulating splines etc...
```

**Comment by whedon** [comment]
[ :point_right: Check article proof :page_facing_up: :point_left: ](https://github.com/openjournals/joss-papers/blob/joss.01731/joss.01731/10.21105.joss.01731.pdf)

**Comment by joelmiller** [comment]
@lpantano @arfon @labarba I've dramatically shortened the paper.  Please let me know if there are any further concerns.

**Comment by joelmiller** [comment]
@whedon check references

**Comment by whedon** [comment]
```
Attempting to check references...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.physa.2006.06.018 is OK
- 10.1016/j.cpc.2017.06.007 is OK
- 10.1103/PhysRevE.96.022301 is OK
- 10.1090/S0002-9947-1945-0013857-4 is OK
- 10.18637/jss.v084.i08 is OK
- 10.1021/j100540a008 is OK
- 10.1103/PhysRevE.93.042316 is OK
- 10.25080/issn.2575-9752 is OK
- 10.1007/978-3-319-50806-1 is OK
- 10.1063/1.5010002 is OK
- 10.1103/PhysRevE.87.060801 is OK
- 10.1093/comnet/cnv021 is OK
- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.1093/bioinformatics/bty921 is OK
- 10.6084/m9.figshare.1164194 is OK
- 10.1103/PhysRevLett.103.058701 is OK
- 10.1186/s13362-019-0058-7 is OK
- 10.1073/pnas.082090499 is OK
- 10.1086/jar.33.4.3629752 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by labarba** [comment]
Could you merge some minor copy edits, here? https://github.com/springer-math/Mathematics-of-Epidemics-on-Networks/pull/47

**Comment by joelmiller** [comment]
merged

**Comment by labarba** [comment]
@whedon accept

**Comment by whedon** [comment]
```
Attempting dry run of processing paper acceptance...
```

**Comment by whedon** [comment]
```Reference check summary:

OK DOIs

- 10.1016/j.physa.2006.06.018 is OK
- 10.1016/j.cpc.2017.06.007 is OK
- 10.1103/PhysRevE.96.022301 is OK
- 10.1090/S0002-9947-1945-0013857-4 is OK
- 10.18637/jss.v084.i08 is OK
- 10.1021/j100540a008 is OK
- 10.1103/PhysRevE.93.042316 is OK
- 10.25080/issn.2575-9752 is OK
- 10.1007/978-3-319-50806-1 is OK
- 10.1063/1.5010002 is OK
- 10.1103/PhysRevE.87.060801 is OK
- 10.1093/comnet/cnv021 is OK
- 10.1098/rsif.2011.0403 is OK
- 10.1103/PhysRevE.80.020901 is OK
- 10.1093/bioinformatics/bty921 is OK
- 10.6084/m9.figshare.1164194 is OK
- 10.1103/PhysRevLett.103.058701 is OK
- 10.1186/s13362-019-0058-7 is OK
- 10.1073/pnas.082090499 is OK
- 10.1086/jar.33.4.3629752 is OK

MISSING DOIs

- None

INVALID DOIs

- None
```

**Comment by whedon** [comment]
Check final proof :point_right: https://github.com/openjournals/joss-papers/pull/1196

If the paper PDF and Crossref deposit XML look good in https://github.com/openjournals/joss-papers/pull/1196, then you can now move forward with accepting the submission by compiling again with the flag `deposit=true` e.g.
 ```
@whedon accept deposit=true
```

**Comment by labarba** [comment]
@whedon accept deposit=true

**Comment by whedon** [comment]
```
Doing it live! Attempting automated processing of paper acceptance...
```

**Comment by whedon** [comment]
🐦🐦🐦 👉 [Tweet for this paper](https://twitter.com/JOSS_TheOJ/status/1208033701399011328) 👈 🐦🐦🐦

**Comment by whedon** [comment]
🚨🚨🚨 **THIS IS NOT A DRILL, YOU HAVE JUST ACCEPTED A PAPER INTO JOSS!** 🚨🚨🚨

 Here's what you must now do:

0. Check final PDF and Crossref metadata that was deposited :point_right: https://github.com/openjournals/joss-papers/pull/1197
1. Wait a couple of minutes to verify that the paper DOI resolves [https://doi.org/10.21105/joss.01731](https://doi.org/10.21105/joss.01731)
2. If everything looks good, then close this review issue.
3. Party like you just published a paper! 🎉🌈🦄💃👻🤘

 Any issues? notify your editorial technical team...

**Comment by labarba** [comment]
Congratulations, @joelmiller, your JOSS paper is published! 🚀 

Huge thanks to our editor: @lpantano, and the reviewers: @acolum, @pholme, @hagberg — we couldn't do this without you and we are grateful 🙏

**Comment by whedon** [comment]
:tada::tada::tada: Congratulations on your paper acceptance! :tada::tada::tada:

If you would like to include a link to your paper from your README use the following code snippets:

```
Markdown:
[![DOI](https://joss.theoj.org/papers/10.21105/joss.01731/status.svg)](https://doi.org/10.21105/joss.01731)

HTML:
<a style="border-width:0" href="https://doi.org/10.21105/joss.01731">
  <img src="https://joss.theoj.org/papers/10.21105/joss.01731/status.svg" alt="DOI badge" >
</a>

reStructuredText:
.. image:: https://joss.theoj.org/papers/10.21105/joss.01731/status.svg
   :target: https://doi.org/10.21105/joss.01731
```
This is how it will look in your documentation:

[![DOI](https://joss.theoj.org/papers/10.21105/joss.01731/status.svg)](https://doi.org/10.21105/joss.01731)

**We need your help!**

Journal of Open Source Software is a community-run journal and relies upon volunteer effort. If you'd like to support us please consider doing either one (or both) of the the following:

- Volunteering to review for us sometime in the future. You can add your name to the reviewer list here: https://joss.theoj.org/reviewer-signup.html
- Making a small donation to support our running costs here: https://numfocus.org/donate-to-joss
